### ISSUE TITLE
Question: Should input clearable button be keyboard navigable?

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-components#2381]
### Question
Currently, the clearable button for an input (when explicitly requested via prop, or with a default clear button like type search), is keyboard navigable. 

Native "clear" functionality does not allow a user to tab to the affordance. It can be clicked but a keyboard user can use `esc` while the input is in focus: 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search

Should we remove the ability to navigate to this button to better match native functionality? All having the button keyboard-navigable does is add more key strokes needed to tab to, and then invoke that clear action, which can already be accomplished while the input is focused.

**Comment by macandcheese** [comment]
Suggested change reduces amount of keystrokes needed for keyboard users to move through input controls. This also would match our "number up / number down" controls - they cannot be focused since the action can be performed in-field.

**Comment by benelan** [comment]
@geospatialem, thoughts on this?

**Comment by geospatialem** [comment]
### Current behavior
Currently the `clearable` property displays a clear button, but is read back to screen readers as "button" without context. We should include some context, such as "clear entry" to provide meaning behind the button, per [WCAG Success Criterion 3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html).

### Discussion on keyboard navigation with clear button on `input`
It is discouraged to not provide all users with access to content, *unless* there is an alternative available to them. However, in this case users could use the `delete` key, `backspace` key, or even select the entire field length and remove the text content.

We probably shouldn't encourage the use of the `esc` key in this case. While `esc` could be used in many cases, it doesn't have full capabilities across OS' and browsers (Android/Firefox and iOS/Safari are known to not get a delete control to use `esc`). For other combinations where `esc` is supported, the control isn't announced. Since its not a common function, usually only power users, trained users, or those that happen upon it through trial and error will use the `esc` key.

### tldr;
We could set the `clearable` button to have a `tabindex="-1"` as long as we add in an `aria-label="clear entry"` to the button. That way if assistive technologies were to discover it, it would be exposed with the proper role and respond via click or keypress events.

**Comment by macandcheese** [comment]
Thanks for the clarity! Just to confirm, this proposal would match the functionality of the "x" in this native example, correct?
![Screen Shot 2022-04-21 at 10 14 15 AM](https://user-images.githubusercontent.com/4733155/164514979-db3fa696-fe20-4c22-a99c-fa113bf4b96a.png)

**Comment by geospatialem** [comment]
That would be a partial support, mostly impacting JAWS users, which is the highest screen reader user (also impacts voice control users).

What are your thoughts on adding a button in with the `aria-label` if text is present? Similar to this [Clear text field buttons example](https://scottaohara.github.io/clear-text-field-button)?

**Comment by macandcheese** [comment]
That makes sense. That's currently the functionality, to only show the "x" when both `clearable` is enabled (or its a type of input that by default has that functionality, to match native), and text is present.

As far as only displaying the "x" when there is focus ... that also sounds good to me and brings us closer to the native implementation, which hides the "x" on blur even when text is present.

**Comment by macandcheese** [comment]
Should we audit the number spinners as well? Those should be closer to the native implementation in that they are unfocusable, but we may want to ensure any changes we make in this case are applied there as well.

**Comment by geospatialem** [comment]
Can you send a resource over, re: number spinner?

**Comment by macandcheese** [comment]
Sure, similarly to how we replace and replicate the native “clear” button - we do the same with number spinners:

https://codepen.io/mac_and_cheese/pen/eYyoWEG

Wanted to make sure they were approached and implemented the same as the clear affordance.

**Comment by geospatialem** [comment]
Ah, thanks - I associate those as "number steppers", thus the confusion on my part. 😕 

Yes, those implementations are solid. 👍🏻 There are some implementations of number steppers where the controls can be accessed via `tab`, but its not necessary.

**Comment by benelan** [comment]
verified on beta.94
